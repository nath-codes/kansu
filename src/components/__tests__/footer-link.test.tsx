import React from 'react'
import {
  cleanup,
  fireEvent,
  getByLabelText,
  render,
} from 'react-testing-library'
import FooterLink from '../footer-link'

describe('FooterLink', () => {
  afterEach(cleanup)

  test('should render correctly', () => {
    const { container } = render(
      <FooterLink ariaLabel="label" href="#">
        <span>bar</span>
      </FooterLink>
    )

    const link = getByLabelText(container, 'label')
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.mouseEnter(link)
    expect(link).toMatchSnapshot()

    fireEvent.mouseLeave(link)
    expect(link).toMatchSnapshot()
  })
})
