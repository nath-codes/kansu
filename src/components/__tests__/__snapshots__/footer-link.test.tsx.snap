// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLink renders correctly 1`] = `
<a
  aria-label="go to foo.com"
  className="sc-bdVaJa bDWFJH"
  href="foo.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "fill": "rgba(73, 85, 94, 1)",
    }
  }
>
  <span />
</a>
`;
